### PR TITLE
Tag resource groups with version info

### LINKED ISSUE #2179: Tag core infra with version information
## Description

As a TRE Administrator/Developer
I want to know which version of the solution is currently deployed on my subscription
So I could plan upgrades and validate deployments

## Acceptance criteria

- [ ] mgmt/devops resource group is tagged
- [ ] core resource group is tagged
- [ ] each deployment updates the tags

### PR DESCRIPTION
**Pull request by tamirkamara**
Fixes #2179 

## What is being addressed

It's hard to know which version of the core infrastructure is currently deployed in an environment.

## How is this addressed

- Include the git source (what repo deployed the environment) and git summary of the last commit, as well as a hard-coded value from a version file.
- Use azcli to update the info on every deployment since we ignore tag info in terraform

## An example of how this looks like
image

### CODE REVIEW

File: templates/core/terraform/update_tags.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
Comment (tanya-borisova) [inline templates/core/terraform/update_tags.sh:1]: Not so keen on this file being an exact copy of `devops/terraform/update_tags.sh`, is it possible to just reference one script?

File: devops/terraform/update_tags.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+mgmt_rg_rid=$(terraform show -json | jq -r '.values.root_module.resources[] | select(.address=="azurerm_resource_group.mgmt") | .values.id')
+echo "Current tags:"
+az tag list --resource-id "${mgmt_rg_rid}"
+
+version=$(cat "${script_dir}"/../version.txt)
+
+# doesn't work with quotes
+# shellcheck disable=SC2206
+version_array=( ${version//=/ } ) # split by =
```
Comment (tanya-borisova) [inline devops/terraform/update_tags.sh:18]: Learned something new today!

**Review by tanya-borisova — COMMENTED** [review]
Wondering if we have to copy paste the script in two locations. Otherwise, looks great!

File: templates/core/terraform/update_tags.sh (hunk 1 above)
Comment (tamirkamara) [reply templates/core/terraform/update_tags.sh:1]: It's not an exact copy as it targets a different resource group...
Most folders have a deploy/destroy script that almost identical to each other.
Never the less, I guess I could move this file to devops/scripts and send the full resource id to update. 
Will that be better @tanya-borisova ?

Comment (tanya-borisova) [reply templates/core/terraform/update_tags.sh:1]: Yeah I'm also not keen on the other almost identical scripts but obviously not asking for it to be fixed in this PR. 
Your idea sounds good, IMO that would be better

Comment (tamirkamara) [reply templates/core/terraform/update_tags.sh:1]: After reconsidering, decided to leave as is

**Review by tanya-borisova — APPROVED** [review]

**Pull request merged**